### PR TITLE
feat: save file

### PR DESCRIPTION
**Pull request by tiste**

### CODE REVIEW

**Comment by tiste** [comment]
Ajouter le menu pour windows/linux.

File: app/components/editor/Editor.tsx [hunk 1]
```diff
@@ -20,13 +23,18 @@ type Props = {
...
     value: { str: string; parsed: object | null; sourceMap: SourceMap | null };
     position: { line: number; column: number } | null;
+    pristine: boolean;
   };
   json: {
     value: { str: string; parsed: object | null; sourceMap: SourceMap | null };
   };
   changeJson: (value: string) => void;
   changeAvro: (value: string) => void;
+  changeAvroPristine: (value: boolean) => void;
```
Comment (delphinaubin) [inline app/components/editor/Editor.tsx]: rename `value` by `isPristine` ?

File: app/components/editor/Editor.tsx [hunk 2]
```diff
@@ -70,78 +78,131 @@ function getPositionFromPath(
...
-                      setErrors(messages);
-                    }}
-                  />
-                </div>
-              </Col>
-              <Col span={12}>
-                <h3 style={{ color: COLORS.DARK_BLUE }}>JSON</h3>
+  saveAvro(path: string | null, value: string) {
+    if (path) {
+      fs.writeFileSync(path, value);
```
Comment (delphinaubin) [inline app/components/editor/Editor.tsx]: I think we should handle fs Errors in the state

Comment (delphinaubin) [inline app/components/editor/Editor.tsx]: the side effect `fs.writeFileSync(path, value)` should be in an action

File: app/containers/EditorPage.tsx [hunk 3]
```diff
@@ -12,7 +16,8 @@ function mapDispatchToProps(dispatch: Dispatch) {
   return bindActionCreators(
     {
       changeAvro: changeAvroWithDispatch,
-      avroMouseMove
+      avroMouseMove,
+      changeAvroPristine: changeAvroPristineWithDispatch
```
Comment (delphinaubin) [inline app/containers/EditorPage.tsx]: I am not sure we need a `changeAcroPristine` action. I think the `changeAvro` could set the `isAvroPristine` state to `true` and the `saveAvroAction` could set it fo `false`

File: app/index.tsx [hunk 4]
```diff
@@ -2,18 +2,27 @@ import fs from 'fs';
...
+  changeAvroWithDispatch
+} from './actions/editor';
 
 const store = configureStore();
 
 ipcRenderer.on('open-file', (_, path) => {
   if (path && fs.existsSync(path)) {
-    const avro = fs.readFileSync(path, 'utf8');
-    changeAvroWithDispatch(avro)(store.dispatch);
+    if (store.getState().editor.avro.pristine) {
```
Comment (delphinaubin) [inline app/index.tsx]: rename `pristine` to `isPristine`

File: app/main-window.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { BrowserWindow } from 'electron';
+
+let mainWindow: BrowserWindow | null = null;
+
+export function setMainWindow(window: BrowserWindow | null) {
```
Comment (delphinaubin) [inline app/main-window.ts]: what is mainWindow ? currently displayed window ?

**Pull request merged**